packages/react-spinner/src/components/Spinner/Spinner.types.ts

```diff
@@ -1,17 +1,55 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { Label } from '@fluentui/react-label';
 
 export type SpinnerSlots = {
-  root: Slot<'div'>;
+  /**
+   * The root of the Spinner.
+   * The root slot receives the `className` and `style` specified directly on the `<Spinner>`.
+   */
+  root: NonNullable<Slot<'div'>>;
+  /**
+   * The slot for the animated svg.
+   * The spinner slot receives the `className` and `style` that handles the spinning animation.
+   * An svg is also rendered as a child of this slot
+   */
+  spinner?: Slot<'span'>;
+  /**
+   * The label of the Slider.
+   * The label slot receives the styling related to the text associated with the Spinner.
+   */
+  label?: Slot<typeof Label>;
 };
 
 type SpinnerCommons = {
-  // TODO Add things shared between props and state here
+  /**
+   * The appearance of the Spinner.
+   * @default 'primary'
+   */
+  appearance?: 'primary' | 'inverted';
+
+  /**
+   * Where the label is positioned relative to the Spinner
+   * @default 'after'
+   */
+  labelPosition?: 'above' | 'below' | 'before' | 'after';
+
+  /**
+   * The size of the spinner.
+   * @default 'medium'
+   */
+  size?: 'tiny' | 'extra-small' | 'small' | 'medium' | 'large' | 'extra-large' | 'huge';
+
+  /**
+   * The status of the Spinner.
+   * @default 'active'
+   */
+  status?: 'active' | 'inactive';
 };
 
 /**
  * Spinner Props
  */
-export type SpinnerProps = ComponentProps<SpinnerSlots> & SpinnerCommons;
+export type SpinnerProps = Omit<ComponentProps<SpinnerSlots>, 'size'> & Partial<SpinnerCommons>;
```

Comment:
I think it is better to have commons have the required/optional based on its usage with SpinnerProps.  Then have SpinnerState make required/optional changes.  This way a caller reading the code doesn't have to do mental gymnastics to figure out what is required/option in props.